### PR TITLE
Updated apache commons text dependency to version 1.10.0

### PR DESCRIPTION
**Pull request by kushagraThapar**
* `azure-cosmos` and `azure-cosmos-encryption` takes test dependency on apache `commons-text`. 
* This PR updates apache commons text dependency to version 1.10.0 in `azure-cosmos` SDK and `azure-cosmos-encryption` SDK
* In addition to test dependency, this PR also updates the shaded version of few classes of apache commons-text to version 1.10.0
* This fixes CVE-2022-42889 - https://www.cve.org/CVERecord?id=CVE-2022-42889, https://commons.apache.org/proper/commons-text/security.html

### CODE REVIEW

**Review by Pilchie — APPROVED** [review]

**Review by joshfree — APPROVED** [review]

**Review by alzimmermsft — APPROVED** [review]

**Review by trande4884 — APPROVED** [review]

File: eng/versioning/external_dependencies.txt [hunk 1]
```diff
@@ -274,7 +274,7 @@ net.oneandone.reflections8:reflections8;0.11.7
 net.jonathangiles.tools:dependencyChecker-maven-plugin;1.0.6
 net.jonathangiles.tools:whitelistgenerator-maven-plugin;1.0.2
 org.apache.commons:commons-collections4;4.4
-org.apache.commons:commons-text;1.6
+org.apache.commons:commons-text;1.10.0
```
Comment (xinlian12) [inline eng/versioning/external_dependencies.txt:277]: will this impact other projects using this library?

Comment (kushagraThapar) [reply eng/versioning/external_dependencies.txt:277]: Only azure-cosmos and azure-cosmos-encryption use this library :)

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**